lib/aspect/k8s/spec.ts

```diff
@@ -0,0 +1,155 @@
+    return `${spec.kind}::${ns}${spec.metadata.name}`;
+}
+
+/**
+ * Return the fingerprint name in format similar to resource name.
+ *
+ * @param fingerprintName internal fingerprint name
+ * @return Human-readable name
+ */
+export function k8sSpecsFingerprintDisplayName(fingerprintName: string): string {
+    return fingerprintName.replace(/::/g, "/");
+}
+
+/**
+ * Read file, parse, and verify it looks like a Kubernetes spec.  If so,
+ * it returns the parse spec object.  If not, it returns `undefined`.
+ */
+export async function parseK8sSpecFile(f: ProjectFile): Promise<K8sObject | undefined> {
+    const specString = await f.getContent();
+    try {
+        const spec = yaml.parse(specString);
+        if (isK8sSpec(spec)) {
+            return spec;
+        }
+    } catch (e) {
+        logger.warn(`Failed to parse ${f.name}, skipping: ${e.message}`);
+    }
+    return undefined;
+}
+
+/**
+ * Test if the provided object looks like a valid Kubernetes resource object.
+ */
+export function isK8sSpec(o: any): o is K8sObject {
+    return o.apiVersion && o.kind && o.metadata && o.metadata.name;
+}
+
+const k8sSpecGlob = "*.@(json|yaml|yml)";
+
+export const createK8sSpecsFingerprints: ExtractFingerprint<FP<K8sObject>> = async p => {
+    const fingerprints: Array<FP<K8sObject>> = [];
+    await projectUtils.doWithFiles(p, k8sSpecGlob, async f => {
+        const spec = await parseK8sSpecFile(f);
+        if (spec) {
+            fingerprints.push(createK8sSpecFingerprint(spec));
+        }
+    });
+    return fingerprints;
+};
+
+export const applyK8sSpecsFingerprint: ApplyFingerprint<FP<K8sObject>> = async (p, fp) => {
+    const specFiles = await projectUtils.toPromise(p.streamFiles(k8sSpecGlob));
+    for (const specFile of specFiles) {
+        const spec = await parseK8sSpecFile(specFile);
+        if (spec && k8sSpecsFingerprintName(spec) === fp.name) {
+            const specString = (/\.ya?ml$/.test(specFile.name)) ? yaml.stringify(fp.data) : stringify(fp.data);
+            await specFile.setContent(specString);
+            return true;
+        }
+    }
+    return false;
+};
+
+export const diffK8sSpecsFingerprints: DiffSummaryFingerprint = (diff, target) => {
+    const resourceName = k8sSpecsFingerprintDisplayName(k8sSpecsFingerprintName(diff.from.data));
+    const repo = `${diff.owner}/${diff.repo}/${diff.branch}`;
+    return {
+        title: "New Kubernetes Spec Policy",
+        description: `Policy version for Kubernetes spec ${bold(resourceName)} differs from that in ${bold(repo)}.`,
+    };
+};
+
+/**
+ * Aspect emitting 0 or more npm dependencies fingerprints.
+ */
+export const NpmDeps: Aspect<FP<K8sObject>> = {
+    displayName: "Kubernetes specs",
+    name: K8sSpecsType,
+    extract: createK8sSpecsFingerprints,
+    apply: applyK8sSpecsFingerprint,
+    summary: diffK8sSpecsFingerprints,
+    toDisplayableFingerprint: fp => fp.data[1],
```

Comment:
`data` is the spec object so array-index access won’t work I think. Realistically this should be a value that can be displayed in a PR body etc.

Reply:
I'm not really sure how to concisely represent the "value" of an entire spec here, but it seems like that is what it wants. That is, the NpmDeps example shows the version there. I've just made it the display name.